### PR TITLE
use PHP 7.4 features and do some cleanup

### PR DESCRIPTION
**Pull request by blizzz**
- removed unused properties
- add property types
- resolve some deprecations
- fix a potential unset variable use

### CODE REVIEW

File: lib/Operation.php [hunk 1]
```diff
@@ -177,6 +165,7 @@ public function onEvent(string $eventName, Event $event, IRuleMatcher $ruleMatch
 							'exception' => $e,
 						]
 					);
+					continue;
```
Comment (nickvergessen) [inline lib/Operation.php:168]: continue at the end of try?
Doesn't look intended?

Comment (come-nc) [reply lib/Operation.php:168]: At the end of catch, to skip in case of PlaceholderNotSubstituted exception, seems legit to me.

**Review by come-nc — APPROVED** [review]

File: lib/Operation.php (hunk 1 above)
Comment (nickvergessen) [reply lib/Operation.php:168]: github indentation threw me off...
The continue is inside the foreach

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**